desktop-widgets/divelistview.cpp

```diff
@@ -120,6 +75,45 @@ DiveListView::~DiveListView()
 	settings.endGroup();
 }
 
+void DiveListView::calculateInitialColumnWidth(const DiveTripModel &tripModel, int col)
+{
+	const QFontMetrics metrics(defaultModelFont());
+	int em = metrics.width('m');
+	int zw = metrics.width('0');
+
+	QString header_txt = tripModel.headerData(col, Qt::Horizontal, Qt::DisplayRole).toString();
+	int width = metrics.width(header_txt);
+	int sw = 0;
+	switch (col) {
+	case DiveTripModel::NR:
+	case DiveTripModel::DURATION:
+		sw = 8*zw;
+		break;
+	case DiveTripModel::DATE:
+		sw = 14*em;
+		break;
+	case DiveTripModel::RATING:
+		sw = static_cast<StarWidgetsDelegate*>(itemDelegateForColumn(col))->starSize().width();
+		break;
+	case DiveTripModel::SUIT:
+	case DiveTripModel::SAC:
+		sw = 7*em;
+		break;
+	case DiveTripModel::PHOTOS:
+		sw = 5*em;
+		break;
+	case DiveTripModel::LOCATION:
+		sw = 50*em;
+		break;
+	default:
+		sw = 5*em;
+	}
+	if (sw > width)
+		width = sw;
+	width += zw; // small padding
+	initialColumnWidths[col] = std::max(initialColumnWidths[col], width);
```

Comment:
`setColumnWidth` and `columnWidths` were added because we had weird out-of-bounds bugs.
the functions had guards.

if all works well a regular array / vector should be fine, of course.

Reply:
This is a private function - nobody else can call it. It's called in the loop
```
for (int i = DiveTripModel::NR; i < DiveTripModel::COLUMNS; i++)
```
and the array is initialized with
```
initialColumnWidths(DiveTripModel::COLUMNS, 50)
```
In my opinion, this is guaranteed to work. Note that DiveTripModel::COLUMNS is a constant (enum) and can't change during the application's lifetime. Also note that initialization and function call is only in the constructor - I think this is really safe.

Reply:
Access to the array is only in the `destructor` and `setupUI`. But there, likewise all is in a loop of the above kind.

So I think the out-of-bound accesses must have had a different cause.